geo/parse.go

```diff
@@ -59,6 +64,35 @@ func ExtractGeoPoint(thing interface{}) (lon, lat float64, success bool) {
 		}
 	}
 
+	// is it a string
+	if thingVal.Kind() == reflect.String {
+		geoStr := thingVal.Interface().(string)
+		if strings.Contains(geoStr, ",") {
+			// geo point with coordinates split by comma
+			points := strings.Split(geoStr, ",")
+			for i, point := range points {
+				// trim any leading or trailing white spaces
+				points[i] = strings.TrimSpace(point)
+			}
+			if len(points) == 2 {
+				var err error
+				lat, err = strconv.ParseFloat(points[0], 64)
+				if err == nil {
+					foundLat = true
+				}
+				lon, err = strconv.ParseFloat(points[1], 64)
+				if err == nil {
+					foundLon = true
+				}
+			}
+		} else {
+			// geo hash
+			lat, lon = geohash.Decode(geoStr)
```

Comment:
worth having a few UTs to test the negative cases for this Decode() fn?,
For eg: like pass couple of invalid hashes,  
Another thing to see is that how that lat/lon values returned on an invalid hash percolates to later sections of the logic.

Reply:
Well..  geohash is a library, that already has it's own suite of unit tests.

Reply:
geohash is a library -> its always a debatable thing about the robustness or maturity of the 3rd party libs.  
Main wrinkle points for me here are,
1. How confident are you that, the values returned from that lib for an invalid hash are valid Or that it won't break any of the remaining code in bleve? 

2. Having few unit tests to set/validate the response expectations from a library helps in confirming minimum sanity at least during a SHA bump for it at a later point in time for accommodating a genuine feature or bug fix on their side.